VPCルーター + F/Wに`terraform state show`実行でクラッシュ

<details>
<summary>エラー内容</summary>
```
$terraform state show sakuracloud_vpc_router_filrewall.send_fw
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x30 pc=0xbc562]

goroutine 1 [running]:
panic(0x2fb1ba0, 0xc8200120c0)
	/opt/go/src/runtime/panic.go:481 +0x3e6
github.com/hashicorp/terraform/command.(*StateShowCommand).Run(0xc8201c7200, 0xc8200720f0, 0x1, 0x1, 0xc82030a0e0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/state_show.go:53 +0x702
github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc8201f4900, 0xc8202119e0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli/cli.go:153 +0x56e
main.wrappedMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:144 +0xaf9
main.realMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:80 +0xa10
main.main()
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:23 +0x28



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```
</detail>

--- a/resource_sakuracloud_vpc_router_firewall.go
+++ b/resource_sakuracloud_vpc_router_firewall.go
@@ -216,9 +216,21 @@ func resourceSakuraCloudVPCRouterFirewallDelete(d *schema.ResourceData, meta int
 		return fmt.Errorf("Couldn't find SakuraCloud VPCRouter resource: %s", err)
 	}
 
-	if vpcRouter.Settings.Router.Firewall != nil {
-		vpcRouter.Settings.Router.Firewall.Config = []*sacloud.VPCRouterFirewallSetting{}
-		vpcRouter.Settings.Router.Firewall.Enabled = "False"
+	direction := d.Get("direction").(string)
+	if vpcRouter.Settings != nil && vpcRouter.Settings.Router != nil && vpcRouter.Settings.Router.Firewall != nil &&
+		vpcRouter.Settings.Router.Firewall.Config != nil {
+
+		switch direction {
+		case "send":
+			vpcRouter.Settings.Router.Firewall.Config[0].Send = nil
+		case "receive":
+			vpcRouter.Settings.Router.Firewall.Config[0].Receive = nil
+		}
+
+		if vpcRouter.Settings.Router.Firewall.Config[0].Send == nil && vpcRouter.Settings.Router.Firewall.Config[0].Receive == nil {
+			vpcRouter.Settings.Router.Firewall.Config = nil
+			vpcRouter.Settings.Router.Firewall.Enabled = "False"
+		}
 
 		vpcRouter, err = client.VPCRouter.UpdateSetting(routerID, vpcRouter)
 		if err != nil {

